learningequality/studio
# Frontend State Refactor, Step 1

## Description

Removes most reliance on `window` bound global state.
Moves a lot of data into autogenerated constant files.
Introduces Vuex state management.
Introduces Jest frontend tests.

## Steps to Test

- Check everything works! Mostly file upload/metadata edit modal
- Also check admin page, as there was a merge conflict on rebase.

#### At a high level, how did you implement this?

Central state object that is importable to track state. Added a Vuex store to this State object for gradual transition of the state to Vuex.

#### Does this introduce any tech-debt items?

Hopefully this is paying off tech debt.


## Checklist

- [ ] Does this introduce a change that needs to be updated in the [user docs](https://kolibri-studio.readthedocs.io/en/latest/index.html)?
